### PR TITLE
Update to recent remoteStorage spec

### PR DESCRIPTION
**Pull request by galfert**
This should implement all missing features to be compatible with spec `draft-dejong-remotestorage-07`.

### CODE REVIEW

File: spec/swift/app_spec.rb [hunk 1]
```diff
@@ -442,6 +468,13 @@ def app
           last_response.headers["ETag"].must_equal "\"f9f85fbf5aa1fa378fd79ac8aa0a457d\""
         end
 
+        it "has a Cache-Control in the header" do
```
Comment (skddc) [inline spec/swift/app_spec.rb]: I think the language can be improved a bit here. E.g. "returns a Cache-Control header" or "includes Cache-Control in the headers" or similar.

File: spec/swift/app_spec.rb [hunk 2]
```diff
@@ -421,6 +429,24 @@ def app
 
       describe "data" do
```
Comment (skddc) [inline spec/swift/app_spec.rb]: This is weird. Should be "documents" according to @galfert.

File: spec/swift/app_spec.rb [hunk 3]
```diff
@@ -421,6 +429,24 @@ def app
...
 
+        it "has the required response headers" do
```
Comment (skddc) [inline spec/swift/app_spec.rb]: `s/has/returns/` maybe?

File: spec/swift/app_spec.rb [hunk 4]
```diff
@@ -473,13 +506,23 @@ def app
...
           last_response.status.must_equal 200
-          last_response.content_type.must_equal "application/json"
+          last_response.content_type.must_equal "application/ld+json"
 
           content = JSON.parse(last_response.body)
           content["items"]["food/"].wont_be_nil
           content["items"]["food/"]["ETag"].must_equal "f9f85fbf5aa1fa378fd79ac8aa0a457d"
         end
 
+        it "responds with 200 and empty object when directory doesn't exist" do
```
Comment (skddc) [inline spec/swift/app_spec.rb]: It's a non-empty object, but with an empty `items` list/key/property. So I guess returns an "empty directory listing" could work. Or the long version. :)

**Comment by skddc** [comment]
Added some comments, but only wording stuff.

**Comment by galfert** [comment]
Improved the specs' wording.

**Pull request merged**